rec: Handle policy (if needed) after postresolve and document the hooks better (#10111)
PowerDNS/pdns: pdns/recursordist/docs/lua-scripting/hooks.rst

Once a script is loaded, PowerDNS looks for the interception functions in the loaded script.
All of these functions are optional.

If ``ipfilter`` returns ``true``, the query is dropped.

> If ``preresolve`` returns ``true``, it will indicate it handled a query, and the recursor will send to result as constructed in the functions to the client.

Reviewer: ```suggestion
If ``preresolve`` returns ``true``, it will indicate it handled a query, and the recursor will send the result as constructed in the function to the client.
```

Once a script is loaded, PowerDNS looks for the interception functions in the loaded script.
All of these functions are optional.

If ``ipfilter`` returns ``true``, the query is dropped.
If ``preresolve`` returns ``true``, it will indicate it handled a query, and the recursor will send to result as constructed in the functions to the client.
If it returns ``false``, the Recursor will continue processing.

> For the other functions, the return value will indicate if DNSSEC validation should take place.

Reviewer: While it's technically the only thing that changes today when `true` is returned, I'm not sure we should document it that way. Perhaps:
```suggestion
For the other functions, the return value will indicate that an alteration has been made. In that case DNSSEC validation will be automatically disabled since the content might not be genuine anymore.
```

> For modified results, the functions should return ``true`` to avoid failing DNSSEC validations.

Reviewer: ```suggestion
```

At specific points the Recursor will check if policy handling should take place.

> These points are immediatly after ``preresolve``, after resolving and after ``postresolve``.

Reviewer: ```suggestion
These points are immediately after ``preresolve``, after resolving and after ``postresolve``.
```

**Warning**: Please do NOT use the above sample script in production!
Responsible NXDomain redirection requires more attention to detail.

> Useful 'rcodes' include 0 for "no error" and ``pdns.NXDOMAIN`` for "NXDOMAIN", ``pdns.DROP`` to drop the question from further processing.

Reviewer: ```suggestion
Useful 'rcodes' include 0 for "no error" and ``pdns.NXDOMAIN`` for "NXDOMAIN". Before 4.4.0, ``pdns.DROP`` can also be used to drop the question without any further processing.
```
